proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+
+  1. If the response was processed successfully, SDL core has to provide the `WARNINGS` result code instead of `SUCCESS`.
+  2. SDL core has to provide removed enum items in `info` string of the response.
+      Since there could be more than one parameter with cut-off value `info` message can be constructed the following way:
+      `Invalid enums were removed: <param_1>:<enum_value_1>,<enum_value_2>;<param_2>:<enum_value_3> ...`
+      If `info` parameter contains other value belonging to the original processing result SDL core has to append information about cut-off enum value(s) to the existing value.
+      
+  **Examples:**
+  In each example below the Mobile Application sends a request to SDL core.
+  1. Request contains a parameter which value is an enum item
+      Request: `SetMediaClockTimer`
+      
+      1.1 Parameter is mandatory: `updateMode="UNKNOWN"`
+      SDL core removes the item `UNKNOWN` from the request.
+      The value of `updateMode` parameter becomes empty.
+      SDL omits empty parameter from the request.
+      Since parameter is mandatory (according to Mobile API), the request fails with the following response to Mobile Application:
+      `success=false, resultCode = "INVALID_DATA", info="Invalid enums were removed: updateMode:UNKNOWN"`
+      
+      1.2 Parameter is optional: `audioStreamingIndicator="UNKNOWN"`
+      SDL core removes the `UNKNOWN` item from the request.
+      The value of `audioStreamingIndicator` parameter becomes empty.
+      SDL core omits the empty parameter in the request.
+      Since the parameter is optional (according to Mobile API), the request is successfully processed with the following response to Mobile Application:
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: audioStreamingIndicator:UNKNOWN"`
+      
+  2. Request contains a parameter which value is an array of enum items
+      Request: `RegisterAppInterface`
+      
+      2.1 Only one enum item is unknown: `appHMIType = [ "DEFAULT", "UNKNOWN"]`
+      SDL core removes the `UNKNOWN` item from the request.
+      The value of `appHMIType` parameter now contains only `DEFAULT` item (which is known one).
+      Request is successfully processed with the following response to Mobile Application:
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: appHMIType:UNKNOWN"`
+      
+      2.2 All enum items are unknown: `appHMIType = [ "UNKNOWN_1", "UNKNOWN_2"]`
+      SDL  core removes both `UNKNOWN_1` and `UNKNOWN_2` items from the request.
+      The value of `appHMIType` parameter becomes empty.
+      Since `minsize` of the parameter is `1` (according to Mobile API), the request fails with the following response to Mobile Application:
+      `success=false, resultCode = "INVALID_DATA", info="Invalid enums were removed: appHMIType:UNKNOWN_1, UNKNOWN_2"`
```

Comment:
```suggestion
      2.2 All enum items are unknown: `appHMIType = [ "UNKNOWN_1", "UNKNOWN_2"]`
      SDL  Core removes both `UNKNOWN_1` and `UNKNOWN_2` items from the request.
      The value of `appHMIType` parameter becomes empty.
      Since `minsize` of the parameter is `1` (according to Mobile API), the request fails with the following response to mobile application:
      `success=false, resultCode = "INVALID_DATA", info="Invalid enums were removed: appHMIType:UNKNOWN_1, UNKNOWN_2"`
```
Please also see comment above regarding line breaks.